src/dmd/dstruct.d

```diff
@@ -227,6 +229,9 @@ extern (C++) class StructDeclaration : AggregateDeclaration
     FuncDeclarations postblits; // Array of postblit functions
     FuncDeclaration postblit;   // aggregate postblit
 
+    CopyCtorDeclaration copyCtor;       // copy constructor
+    bool[ModBits] copyCtorTypes;        // source-destination qualifiers for the struct copy constructors
```

Comment:
Is there any way to use D's hashtable here and translate it to the C++ headers somehow? Or should I work around it?
Does making the variable private and declare public methods that operate on the hashtable work?

Reply:
The bultin-hashtables do not have a ABI gurantee. Therefore avoid them in datastructures part which are part of the AST-Interface.

Reply:
Is that piece of data required to be accessed from the glue layer? if not `void*` will do.

Reply:
See https://github.com/dlang/dmd/blob/aa9411b9b59fcd9746ebf93467f33f50e26933e2/src/dmd/template.h#L65

We already have a precedence for having hash tables in AST.

Reply:
Thanks!